sql-cli/sql_cli/sql_directory_parser.py

```diff
@@ -110,8 +111,25 @@ def get_relative_target_path(self) -> Path:
 
         return target_path.relative_to(self.target_directory)
 
+    def to_transform_operator(self) -> TransformOperator:
+        """
+        Converts SQLFile into a TransformOperator that can be added to a DAG.
+        Any relevant metadata from the file frontmatter will be passed to the TransformOperator,
+        though we do not pass parameter dependencies at this stage.
+        :return: a TransformOperator
+        """
+        return TransformOperator(
+            conn_id=self.metadata.get("conn_id"),
+            parameters=None,
```

Comment:
Why are we not adding parameters here, like how you did it above? 🤔 Is it too complex?
```python
param_dict[s.path.stem].parameters[p] = param_dict[p].output
```

Reply:
Because the parameters don't exist yet. We have to create all of the transform operators and THEN add them as parameters/set dependencies.

Reply:
Ah makes sense 👍